src/foam/nanos/notification/NotificationSetting.js

```diff
@@ -44,10 +46,31 @@ foam.CLASS({
     {
       class: 'Long',
       name: 'id'
+    },
+    {
+      class: 'Boolean',
+      name: 'enabled',
+      value: true
     }
   ],
 
   methods: [
+    {
+      name: 'doNotify',
+      args: [
+        { name: 'x', type: 'Context' },
+        { name: 'user', type: 'foam.nanos.auth.User' },
+        { name: 'notification', type: 'foam.nanos.notification.Notification' }
+      ],
+      javaCode: `
+        // Disable email sending when performing a doNotify to prevent duplicate emails
+        notification = (Notification) notification.fclone();
+        notification.setEmailIsEnabled(false);
```

Comment:
Just noticing this here, but why do we need medium/message specific flags on the notification model now that we have notificationSettings?

Reply:
legacy thing .. i'll remove it .. it'll break nanopay though without a merge there too, so i'll have to prepare a PR on that branch as well.